directory/user/domain/objects.py

```diff
@@ -23,3 +27,29 @@ def by_email(self, email, password):
             raise UserWithSpecifiedEmailAddressAlreadyExistsError
 
         self.user.create_with_email(email=email, password=password)
+
+
+class ChangeUserPassword:
+    """
+    User password implementation.
+    """
+
+    def __init__(self, user):
+        """
+        Constructor.
+        """
+        self.user = user
+
+    def do(self, email, old_password, new_password):
+        """
+        Change user password.
+        """
+        if not self.user.does_exist(email=email):
+            raise UserWithSpecifiedEmailAddressDoesNotExistError
+
+        is_password_matched = self.user.verify_password(email=email, password=old_password)
+
+        if is_password_matched is False:
```

Comment:
`if is_password_matched is False` > `if not is_password_matched`.

Reply:
@dmytrostriletskyi, done.